app/controllers/conference_controller.rb

```diff
@@ -15,6 +15,9 @@ def schedule
     @rooms = @conference.venue.rooms if @conference.venue
     @events = @conference.program.events
     @dates = @conference.start_date..@conference.end_date
+    confirmed_events = @conference.program.events.where(state: 'confirmed')
+    @scheduled_events = confirmed_events.where('start_time IS NOT NULL').order(:start_time)
```

Comment:
Also in the event model, an event is scheduled if it has not only start_time, but also room. So I think that if the room is necessary in the schedule model it should also be necessary in the program scope. In this case I would introduce that it is confirmed too.